src/hera/workflows/v5/_mixins.py

```diff
@@ -45,23 +37,42 @@
 )
 from hera.workflows.v5.env import _BaseEnv
 from hera.workflows.v5.env_from import _BaseEnvFrom
-from hera.workflows.v5.operator import Operator
 from hera.workflows.v5.parameter import Parameter
 from hera.workflows.v5.resources import Resources
-from hera.workflows.v5.task_result import TaskResult
 from hera.workflows.v5.user_container import UserContainer
 from hera.workflows.v5.volume import _BaseVolume
-from hera.workflows.v5.workflow_status import WorkflowStatus
 
 Inputs = List[Union[ModelInputs, Parameter, ModelParameter, Artifact]]
 Outputs = List[Union[ModelOutputs, Parameter, ModelParameter, Artifact]]
 TSub = TypeVar("TSub", bound="_SubNodeMixin")
+TContext = TypeVar("TContext", bound="_ContextMixin")
 
 
 class _BaseMixin(_BaseModel):
     pass
 
 
+class _ContextMixin(_BaseModel):
+    def __enter__(self: TContext) -> TContext:
+        """Enter the context of the workflow"""
+        from hera.workflows.v5._context import _context
+
+        _context.enter(self)
+        return self
+
+    def __exit__(self: TContext, exc_type, exc_val, exc_tb) -> None:
+        """Leave the context of the workflow.
+
+        This supports using `with Workflow(...)`.
+        """
+        from hera.workflows.v5._context import _context
+
+        _context.exit()
+
+    def _add_sub(self, node: Any) -> Any:
+        raise NotImplementedError()
+
```

Comment:
Related to https://github.com/argoproj-labs/hera-workflows/pull/449#discussion_r1125716579